10/README.md

```diff
@@ -0,0 +1,88 @@
+- It is divided into different subsequent steps.
+- Every step is self contained and outputs a usable system.
+- Every step should be a unit of work reasonably small.
+- It is not a requirement to fully migrate the system, the rewrite can be stopped at any point.
+
+From a bird's-eye perspective, the idea is to replace the Tendermint process BigchainDB depends on, with our own process: `bigchaindb-go` (this is not the final name). The new `bigchaindb-go` will allow a tighter implementation with Tendermint Core. The goal is to incrementally replace the HTTP API endpoints implemented in Python with new ones implemented in `bigchaindb-go`.
+
+This approach is not new. Martin Fowler defined this approach in 2004, in his blog post [Strangler Application][strangler:application]. A [paper has been presented][strangler:paper] the same year, during a conference on extreme programming (XP2004). Benefits of this approach, compared to other approaches like _big bang refactoring_, has been proven valid by many [case][strangler:case-study-1] [studies][strangler:case-study-2].
+
+### Step 1: Posting a transaction
+Posting a transaction to BigchainDB is one of the main primitives of the system.
+
+The goal of _Step 1_ is to expose through `bigchaindb-go` one single endpoint fully compatible with [`POST /api/v1/transactions`][bdb:post-tx], and implement the logic required to satisfy the [ABCI][abci] interface.
+
+The integration between the existing BigchainDB Server code (written in Python) and the new `bigchaindb-go` is described in the following section.
+
+About the existing BigchainDB Server:
+- All the HTTP API endpoints will still be served from the existing code.
+- The endpoint `POST /api/v1/transactions` will be just a _pass through proxy_ to the `bigchaindb-go` implementation.
+- A new endpoint `POST /api/v1/validate` will be implemented, to check the validity of a transaction (this **temporary** endpoint will be used by `bigchaindb-go` to do the actual validation, and will give us more time to actually focus on the experiment).
+
+About the new `bigchaindb-go` Server:
+- It implements the endpoint `POST /api/v1/transactions`.
+- It implements the ABCI interface (`check_tx`, `begin_block`, `deliver_tx`, `end_block`, `commit`), and communicates to the local MongoDB instance to write transactions once they are validated, ideally using the [in process interface][abci:in-process].
+
+The following diagram shows how the two systems work together.
+
+![Interaction between the existing BigchainDB Server and bigchaindb-go][diagram]
+
+It is important to notice that the existing BigchainDB Server will only **read** from the shared MongoDB instance, while `bigchaindb-go` will be the only process able to **write** data in the underlying database.
+
+### Other steps
+If the experiment is successful, other API endpoints can be easily integrated one by one into the `bigchaindb-go` implementation. This BEP does not cover this part for now.
```

Comment:
I wonder if we ever need to translate the other endpoints into Golang. They're mostly for doing special queries via GET requests. I've been writing about how node operators can expose their local MongoDB database to the outside world, and their options for doing so. They _could_ just to expose MongoDB on port 27017 (to authenticated users with limited access permissions, such as read-only). But a node operator could also expose an HTTP API or a GraphQL API to their MongoDB database, for example. The "HTTP API for Querying BigchainDB" could be a _stand alone_ program / executable / package. It would be up to the node operator to decide what to expose to external users, and how.

Reply:
Yep, what you are saying makes total sense. This BEP just mention that integration is easy, but does not actively promote it.